pkg/gofr/datasource/pubsub/nats/stream_manager.go

```diff
@@ -0,0 +1,91 @@
+package nats
+
+import (
+	"context"
+	"errors"
+
+	"github.com/nats-io/nats.go/jetstream"
+	"gofr.dev/pkg/gofr/datasource/pubsub"
+)
+
+type StreamManager struct {
+	js     jetstream.JetStream
+	logger pubsub.Logger
+}
+
+func NewStreamManager(js jetstream.JetStream, logger pubsub.Logger) *StreamManager {
+	return &StreamManager{
+		js:     js,
+		logger: logger,
+	}
+}
+
```

Comment:
It will be nice if we can add godoc comment for these methods as they are useful

Reply:
added